ERCS/erc-7760.md

```diff
@@ -0,0 +1,511 @@
+            bytes32(0x3d6000803e603e573d6000fd5b3d6000f35b6020600f3d393d51543d52593df3),
+            args
+        );
+    }
+
+    function deploy(address implementation, bytes memory args)
+        internal
+        returns (address instance)
+    {
+        bytes memory m = initCode(implementation, args);
+        assembly {
+            instance := create(0, add(m, 0x20), mload(m))
+        }
+        require(instance != address(0), "Deployment failed.");
+    }
+}
+```
+
+### Minimal ERC-1967 beacon proxy implementation
+
+#### Minimal ERC-1967 beacon proxy implementation (regular variant)
+
+```solidity
+pragma solidity ^0.8.0;
+
+library ERC1967MinimalBeaconProxyLib {
+    function initCode(address beacon, bytes memory args) internal pure returns (bytes memory) {
+        uint256 n = 0x0052 + args.length;
+        require(n <= 0xffff, "Immutable args too long.");
+        return abi.encodePacked(
+            bytes1(0x61),
+            uint16(n),
+            bytes7(0x3d8160233d3973),
+            beacon,
+            bytes23(0x60195155f3363d3d373d3d363d602036600436635c60da),
+            bytes32(0x1b60e01b36527fa3f0ad74e5423aebfd80d3ef4346578335a9a72aeaee59ff6c),
+            bytes32(0xb3582b35133d50545afa5036515af43d6000803e604d573d6000fd5b3d6000f3),
+            args
+        );
+    }
+
+    function deploy(address beacon, bytes memory args) internal returns (address instance) {
+        bytes memory m = initCode(beacon, args);
+        assembly {
+            instance := create(0, add(m, 0x20), mload(m))
+        }
+        require(instance != address(0), "Deployment failed.");
+    }
+}
+```
+
+#### Minimal ERC-1967 beacon proxy implementation (I-variant)
+
+```solidity
+pragma solidity ^0.8.0;
+
+library ERC1967IMinimalBeaconProxyLib {
+    function initCode(address beacon, bytes memory args) internal pure returns (bytes memory) {
+        uint256 n = 0x0057 + args.length;
+        require(n <= 0xffff, "Immutable args too long.");
+        return abi.encodePacked(
+            bytes1(0x61),
+            uint16(n),
+            bytes7(0x3d8160233d3973),
+            beacon,
+            bytes28(0x60195155f3363d3d373d3d363d602036600436635c60da1b60e01b36),
+            bytes32(0x527fa3f0ad74e5423aebfd80d3ef4346578335a9a72aeaee59ff6cb3582b3513),
+            bytes32(0x3d50545afa361460525736515af43d600060013e6052573d6001fd5b3d6001f3),
+            args
+        );
+    }
+
+    function deploy(address beacon, bytes memory args) internal returns (address instance) {
+        bytes memory m = initCode(beacon, args);
+        assembly {
+            instance := create(0, add(m, 0x20), mload(m))
+        }
+        require(instance != address(0), "Deployment failed.");
+    }
+}
+```
+
+## Security Considerations
+
+### Transparent upgradeable proxy factory security considerations
+
+The transparent upgradeable proxy factory MUST implement proper access control to allow proxies to be upgraded by only authorized accounts.
```

Comment:
MUST is all caps creates a feel like this is still specification because of RFC 2119. It'd suggest you change the tone
```suggestion
To ensure security, the transparent upgradeable proxy factory must implement proper access control to allow proxies to be upgraded by only authorized accounts.
```